rhash and utf8proc failed to build because of missing libcrypto.1.0.0.dylib

Hi!

first of all: Thanks for the great work and the many updates you did over the last few weeks! :-)

Unfortunately I ran into a problem while upgrading my installed stuff:

When I tried to upgrade **cmake** it needs to install **rash** as a dependency.
This fails because of the following eror:

Library not loaded: /usr/local/lib/libcrypto.1.0.0.dylib
  Referenced from: /usr/local/opt/ld64/bin/ld
  Reason: image not found  

It is the same with **utf8proc** which is a dependency to **tmux**

I am running a iMac G3 with 10.4.11 Tiger.

Do you have an Idea what my be the problem here?
Below I postet the terminal outputs of both builds.

 Thanks for your help!

**Trying to build rhash:** 

imac-g3:~ jan$ brew install rhash
==> Downloading https://downloads.sourceforge.net/project/rhash/rhash/1.3.3/rhash-1.3.3-src.tar.gz
Already downloaded: /Users/jan/Library/Caches/Homebrew/rhash-1.3.3.tar.gz
==> Patching
patching file librhash/Makefile
Hunk #1 succeeded at 28 (offset 2 lines).
Hunk #2 succeeded at 182 (offset 6 lines).
==> make lib-static lib-shared all CC=/usr/local/bin/gcc-4.2
Last 15 lines from /Users/jan/Library/Logs/Homebrew/rhash/01.make:
/usr/local/bin/gcc-4.2 -c  -pipe -DIN_RHASH  -O2 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations tth.c -o tth.o
/usr/local/bin/gcc-4.2 -c  -pipe -DIN_RHASH  -O2 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations torrent.c -o torrent.o
/usr/local/bin/gcc-4.2 -c  -pipe -DIN_RHASH  -O2 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations whirlpool.c -o whirlpool.o
/usr/local/bin/gcc-4.2 -c  -pipe -DIN_RHASH  -O2 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations whirlpool_sbox.c -o whirlpool_sbox.o
ar cqs librhash.a algorithms.o byte_order.o plug_openssl.o rhash.o rhash_timing.o rhash_torrent.o util.o aich.o crc32.o ed2k.o edonr.o hex.o md4.o md5.o sha1.o sha256.o sha512.o sha3.o ripemd-160.o gost.o has160.o snefru.o tiger.o tiger_sbox.o tth.o torrent.o whirlpool.o whirlpool_sbox.o
ranlib: file: librhash.a(plug_openssl.o) has no symbols
ranlib: file: librhash.a(util.o) has no symbols
make -C librhash lib-shared
/usr/local/bin/gcc-4.2 -fpic  -pipe -DIN_RHASH  -O2 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations -dynamiclib algorithms.c byte_order.c plug_openssl.c rhash.c rhash_timing.c rhash_torrent.c util.c aich.c crc32.c ed2k.c edonr.c hex.c md4.c md5.c sha1.c sha256.c sha512.c sha3.c ripemd-160.c gost.c has160.c snefru.c tiger.c tiger_sbox.c tth.c torrent.c whirlpool.c whirlpool_sbox.c   -Wl,-install_name,/usr/local/lib/librhash.0.dylib -o librhash.0.dylib
collect2: ld terminated with signal 5 [Trace/BPT trap]
**dyld: Library not loaded: /usr/local/lib/libcrypto.1.0.0.dylib
  Referenced from: /usr/local/opt/ld64/bin/ld
  Reason: image not found**
make[1]: *** [librhash.0.dylib] Error 1
make: *** [librhash/librhash.so.0] Error 2

READ THIS: https://github.com/mistydemeo/tigerbrew/wiki/troubleshooting

---------------------------------------------------------------------------------------------------------------------------------

**Trying to build utf8proc:**

imac-g3:~ jan$ brew install utf8proc
==> Downloading https://github.com/JuliaStrings/utf8proc/archive/v2.8.0.tar.gz
Already downloaded: /Users/jan/Library/Caches/Homebrew/utf8proc-2.8.0.tar.gz
==> Patching
patching file Makefile
==> make install prefix=/usr/local/Cellar/utf8proc/2.8.0
Last 15 lines from /Users/jan/Library/Logs/Homebrew/utf8proc/01.make:
2023-08-24 07:45:43 +0200

make
install
prefix=/usr/local/Cellar/utf8proc/2.8.0

/usr/local/bin/gcc-4.2 -I/usr/local/opt/ld64/include -F/usr/local/Frameworks -Os -w -pipe -mcpu=750 -faltivec -mmacosx-version-min=10.4 -fPIC -std=c99 -Wall -Wextra -pedantic -DUTF8PROC_EXPORTS  -c -o utf8proc.o utf8proc.c
rm -f libutf8proc.a
ar crs libutf8proc.a utf8proc.o
/usr/local/bin/gcc-4.2 -L/usr/local/opt/ld64/lib -L/usr/local/lib -F/usr/local/Frameworks -Wl,-headerpad_max_install_names -B/usr/local/Cellar/ld64/97.17/bin/ -dynamiclib -o libutf8proc.2.dylib utf8proc.o -install_name /usr/local/Cellar/utf8proc/2.8.0/lib/libutf8proc.2.dylib -Wl,-compatibility_version -Wl,2 -Wl,-current_version -Wl,2.6.0
collect2: ld terminated with signal 5 [Trace/BPT trap]
**dyld: Library not loaded: /usr/local/lib/libcrypto.1.0.0.dylib
  Referenced from: /usr/local/Cellar/ld64/97.17/bin/ld
  Reason: image not found**
make: *** [libutf8proc.2.dylib] Error 1

READ THIS: https://github.com/mistydemeo/tigerbrew/wiki/troubleshooting

imac-g3:~ jan$

--- a/Library/Formula/ld64.rb
+++ b/Library/Formula/ld64.rb
@@ -8,12 +8,7 @@ class Ld64 < Formula
   #
   url "https://github.com/apple-oss-distributions/ld64/archive/refs/tags/ld64-97.17.tar.gz"
   sha256 "dc609d295365f8f5853b45e8dbcb44ca85e7dbc7a530e6fb5342f81d3c042db5"
-
-  bottle do
-    sha1 "5a42d849b7231d0b5985514ed5e77aa424048853" => :tiger_altivec
-    sha1 "9c32cb4c189dd8c05f5fed54eb05cf7b2220653d" => :leopard_g3
-    sha1 "28728a210881f5b6949c3305e932eed9109b3a10" => :leopard_altivec
-  end
+  revision 1
 
   resource "makefile" do
     url "https://trac.macports.org/export/123511/trunk/dports/devel/ld64/files/Makefile-97", :using => :nounzip
@@ -27,6 +22,8 @@ class Ld64 < Formula
   depends_on "cctools-headers" => :build
   depends_on "dyld-headers" => :build
   depends_on "libunwind-headers" => :build
+  # No CommonCrypto
+  depends_on "openssl" if MacOS.version < :leopard
 
   keg_only :provided_by_osx,
     "ld64 is an updated version of the ld shipped by Apple."

